src/plugins/sync_to_async.ts

```diff
@@ -0,0 +1,380 @@
+      // use that binding to replace an expression.
+      if (id) {
+        // If names has any values, then we need to replace a wrapper expression
+        // direct property access, i.e. instead of `this.ticket().requester().email()`
+        // we would replace with `_ticket.requester.email` (where `_ticket` was a new
+        // hoisted variable we created).
+        if (names.length) {
+          // Just concat the binding name with property names
+          exp.replaceWithSourceString(`${id.name}.${names.join(".")}`);
+          // If this expression is itself an assignment to a binding, we need
+          // to continue processing further references, i.e.
+          // `var ticket = this.ticket(); var requester = ticket.requester();`
+          if (exp.parentPath.isVariableDeclarator()) {
+            const binding = opCache.scope.bindings[exp.parent.id.name];
+            replaceReferencesForBinding(binding);
+          }
+        } else {
+          // If we get to here, we're just using the value of the binding
+          // directly, i.e. `return this.ticket()`.
+          exp.replaceWith(id);
+        }
+        toAsync = true;
+      }
+    }
+    // If we didn't set `toAsync`, then we haven't found/replaced any expressions.
+    if (!toAsync) return;
+    // If `asyncMethods` exists and this app method isn't in the queue to be processed
+    // in the second pass, add it, since it will become async.
+    if (asyncMethods && !asyncMethods.includes(opName)) {
+      asyncMethods.push(opName);
+    }
+    // Set the FunctionExpression node to be async.
+    op.node.value.async = true;
+    // Keep a reference to the method being async in metadata for lookup later
+    opCache.async = true;
+  }
+};
+
+interface IManifest {
+  [key: string]: any; // number | object | string | boolean
+  location?: string | string[];
+  noTemplate?: string | string[];
+}
+
+/**
+ * Checks whether the given location is specified in the manifest.
+ * A location could appear in either the `location`, or the `noTemplate`.
+ * 
+ * @param {IManifest} manifest 
+ * @param {string} name 
+ * @returns {boolean} 
+ */
+function hasLocation(manifest: IManifest, name: string): boolean {
+  // Store the intersection of all the possible locations in a
+  // new transient property on the manifest object.
+  if (!manifest.allLocations) {
+    let { location, noTemplate } = manifest;
+    manifest.allLocations = chain([])
+      .concat(location, noTemplate)
+      .compact()
+      .uniq();
+  }
+  return manifest.allLocations.some((loc: string) =>
+    new RegExp(name).test(loc)
+  );
+}
+
+export default (json, path, cache) => {
+  // Check whether this app cares about APIs in certain
+  // locations.  This way, we only need to check for relevant
+  // method expressions when rewriting the AST.
+  const hasTicket = hasLocation(json, "ticket");
+  const hasUser = hasLocation(json, "user");
+  const hasOrg = hasLocation(json, "organization");
```

Comment:
No `top_bar` or `nav_bar`? i.e. `preloadPane`, `popover`, though it may not be relevant until we support `invoke`.

Reply:
As discussed in 🍖 -space, I'll leave the `*bar` locations and invoke for a follow up PR